components/01-atoms/images/icons/_icon.twig

```diff
@@ -0,0 +1,39 @@
+{#
+ # Available variables:
+ # - icon__name: the name of the icon
+ # - icon__role: a11y role
+ # - icon__title: a11y title
+ # - icon__desc: a11y description
+#}
+
+{% set icon__base_class = icon__base_class|default('icon') %}
+{# If `directory` is defined, set the path relative for Drupal.
+ # Otherwise, set the path relative to the Component Library. #}
+{% set icon__url = directory ? '/' ~ directory ~ '/dist/' %}
+
+{% set unique_name = icon__name ~ '-' ~ random() %}
+
+<svg {{ bem(icon__base_class, icon__modifiers, icon__blockname, icon__extra_class) }}
+
+  {% if icon__decorative %}
+    aria-hidden="true" role="img"
+  {% elseif icon__role %}
+    role="{{ icon__role }}"
```

Comment:
I want to make sure I understand: `icon__role` is a setting from Drupal? Either the icon is decorative `icon__decorative` or is another role (img, document or application are all applicable to svgs). 

Application and Document roles would be for complex svgs, like interactive graphs- is it safe to set any other icon that is not decorative to `role="img"`?

Reply:
I figured out how those variables are set, icon__role and icon__decorative!

Reply:
Yep. I think you have it all worked out, but just for posterity, these variables are manually set upon implementation. As you've seen already, I need to add `icon__decorative` to most (if not all) of our icons. But if one needed another role, for whatever reason, we could add it using one of these variables